app/concepts/todo/create.rb

```diff
@@ -1,50 +1,38 @@
 class Todo::Create < Trailblazer::Operation
+  include Model
+  model Todo, :create
+
+  contract do
+    property :title, validates: { presence: true }
+  end
 
-  # Logic for creating a TODO
-  #     if `todo_list` is specified
-  #         create the `todo`
-  #         associated with `todo_list`
-  #     else
-  #         create the `todo`
-  #         create a `user`
-  #         create a default `todo_list`
-  #         associate `todo_list` with `user`
-  #         associate `todo` with `todo_list`
   def process(params)
-    # make sure title is not blank
-    return invalid! if params[:todo][:title].blank?
+    validate(params[:todo]) do
+      contract.model.list = get_or_create_todo_list(params)
+      contract.model.list.user = get_or_create_user(params)
+      contract.model.list.save
+      contract.save
+    end
+  end
```

Comment:
Yup. Simply move it to a different method inside your operation. Who knows in the future it can be moved to an external object that can be used from anywhere else and not just inside your operation.